### ISSUE TITLE
Hard to converge

### ISSUE THREAD

**Issue by squidszyd** [weiliu89/caffe#187]
I'm working on detection relatively small objects but encounter with convergence problem.
I've modified the scale parameters in multi-box generation to let the network adapt to small objects.
At the beginning, the loss keep rising and finally get to 'nan'. I tried to decrease the learning rate but still hard to converge. The loss is always greater than 1.0 while the 'mbox-loss' sometimes goes below 1.0.
Is SSD incapable of detecting small objects in large images ( 40 x 100 in 1448 x 888 ) or is the training set(about 3400 images) too small?

**Comment by weiliu89** [comment]
As long as the loss is going down, then it should be good. What do you expect it to converge? The loss won't go down to 0 (if it does, it probably overfit to your dataset pretty badly). You could try to evaluate or visualize your model outputs to see if it makes sense or not.
